pandas/core/generic.py

```diff
@@ -842,7 +842,25 @@ def __contains__(self, key):
 
     @property
     def empty(self):
-        """True if NDFrame is entirely empty [no items]"""
+        """True if NDFrame is entirely empty [no items], i.e. any of the axes
+        are of length 0.
+
+        Notes
+        -----
+        If NDFrame contains only NaNs, it is still not considered empty.
+
+        Examples
+        --------
+
+        >>> # containing only NaNs does not make the df empty
+        >>> df = pd.DataFrame({'A' : [np.nan]})
+        >>> df.empty
+        False
+        >>> # if we drop NAs, the axes are now of length 0
+        >>> df.dropna().empty
```

Comment:
Idem as above about the comment + I would also show the output of the actual empty frame